device/src/esp32_run.rs

```diff
@@ -61,34 +61,31 @@ where
             mut sha256,
         } = self;
 
+        let mut keygen = HmacKeyGen::new();
         let flash = FlashStorage::new();
         let mut flash = storage::DeviceStorage::new(flash);
         let ota_config = ota::OtaConfig::new(flash.flash_mut());
         let active_partition = ota_config.active_partition(flash.flash_mut());
         let active_firmware_digest = active_partition.digest(flash.flash_mut(), &mut sha256);
         ui.set_busy_task(ui::BusyTask::Loading);
 
-        let (mut signer, mut name, mut key_names) =
+        let (mut signer, mut name) =
             match flash.read_header().expect("failed to read header from nvs") {
                 Some(header) => {
                     let mut signer = FrostSigner::new(KeyPair::<Normal>::new(header.secret_key));
                     let mut name: Option<alloc::string::String> = None;
-                    let mut key_names: BTreeMap<KeyId, String> = BTreeMap::new();
 
                     for change in flash.iter() {
                         match change {
-                            storage::Change::Core(core) => {
-                                signer.apply_change(core);
+                            storage::Change::Core(mutation) => {
+                                signer.apply_mutation(&mutation);
                             }
                             storage::Change::Name(name_update) => {
                                 name = Some(name_update);
                             }
-                            storage::Change::KeyNamed((key_id, key_name)) => {
-                                key_names.insert(key_id, key_name);
-                            }
                         }
                     }
-                    (signer, name, key_names)
+                    (signer, name)
```

Comment:
This was weird. `key_names` was being stored here even though (after much discussion) they were meant to be in `frostsnap_core`'s state. Looks like we forgot to do this in an earlier PR.